src/liquid/components/ld-file-upload/ld-file-upload.tsx

```diff
@@ -0,0 +1,1034 @@
+      )
+    }
+
+    this.uploadItems = [
+      ...this.uploadItems.slice(0, itemToDeleteIndex),
+      ...this.uploadItems.slice(itemToDeleteIndex + 1),
+    ]
+  }
+
+  @Watch('uploadItems')
+  async changeProgressVisualisation() {
+    if (this.uploadItems.length == 0) {
+      this.startUploadClicked = false
+    }
+    if (
+      this.uploadItems.length == 0 &&
+      this.renderOnlyChooseFile == false &&
+      this.cannotBeChosen.length == 0 &&
+      this.exceedMaxSize.length == 0
+    ) {
+      this.renderOnlyChooseFile = true
+      this.startUploadClicked = false
+      this.renderCircularProgress = false
+    }
+    if (
+      this.uploadItems.filter((item) => item.state == 'uploaded').length ==
+        this.uploadItems.length &&
+      this.uploadItems.length != 0
+    ) {
+      this.allUploadsFinished = true
+    }
+
+    if (
+      this.uploadItems.filter((item) => item.state == 'uploaded').length ==
+        this.uploadItems.length &&
+      this.uploadItems.length != 0 &&
+      this.circularProgress
+    ) {
+      console.log({
+        uploadedItems:
+          this.uploadItems.filter((item) => item.state == 'uploaded').length ==
+          this.uploadItems.length,
+        uploadItemslength: this.uploadItems.length != 0,
+        cicularProgress: this.circularProgress,
+      })
+      this.allUploadsFinished = true
+      const delay = (ms) => new Promise((res) => setTimeout(res, ms))
+      await delay(5000)
+      this.uploadItems = []
+      this.renderOnlyChooseFile = true
+      this.startUploadClicked = false
+      this.renderCircularProgress = false
+      this.allUploadsFinished = false
+    }
+  }
+
+  private bytesToSize = (bytes: number) => {
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB']
+
+    let sizeIndex = 0
+
+    while (bytes >= 1024 && sizeIndex < sizes.length - 1) {
+      bytes /= 1024
+      sizeIndex++
+    }
+
+    const roundedSize = Number(bytes.toFixed(2))
+
+    return roundedSize + ' ' + sizes[sizeIndex]
+  }
+
+  private calculateTotalProgress = () => {
+    const activeUploads = this.uploadItems.filter(
+      (item) =>
+        item.state == 'pending' ||
+        item.state == 'paused' ||
+        item.state == 'uploading' ||
+        item.state == 'uploaded'
+    )
+    console.log(activeUploads)
+    const totalSizeSum = activeUploads.reduce(
+      (partialSum, file) => partialSum + file.fileSize,
+      0
+    )
+    const uploadedSizeSum = activeUploads.reduce(
+      (partialSum, file) => partialSum + file.fileSize * (file.progress / 100),
+      0
+    )
+    const totalProgress = uploadedSizeSum / totalSizeSum || 0
+    console.log('totalProgress', totalProgress)
+    // returns total progress between 0 and 1
+    return totalProgress
+  }
+
+  private addChosenFiles(chosenFiles: UploadItem[]) {
+    for (let i = 0; i < chosenFiles.length; i++) {
+      this.allChosenFiles.push(chosenFiles[i])
+    }
```

Comment:
Maybe this will do the job as well. Not sure though. Please test.
```suggestion
    this.allChosenFiles = [...this.allChosenFiles, ...chosenFiles]
```